content/articles/2017-07-ggplot2-3-0-0-tidy-evaluation.Rmarkdown

```diff
@@ -0,0 +1,116 @@
+---
+title: ggplot2 3.0.0 - tidy evaluation
+date: '2018-07-11'
+slug: ggplot2-3-0-0-tidy-evaluation
+author: Mara Averick
+categories: [package]
+description: >
+  Changes in ggplot2 3.0.0, tidy-evaluation edition.
+photo:
+  url: https://unsplash.com/photos/8KfCR12oeUM
+  author: Christopher Burns
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>", fig.retina = 2, dpi = 300)
+library(tidyverse)
+```
+
+## Tidy evaluation
+
+One of the biggest changes in [ggplot2](https://ggplot2.tidyverse.org/) 3.0.0 is support for [tidy evaluation](https://adv-r.hadley.nz/evaluation.html#tidy-evaluation), making it more programmable, and more consistent with the rest of the tidyverse. Though this introduces some breaking changes, we believe it to be worthwhile in the interest of improving future code. Here, we outline some of the more prominent of those changes, and "symptomatic" error messages you may encounter. For our complete list of such messages, please see the [Breaking changes](https://ggplot2.tidyverse.org/news/index.html#breaking-changes) section of the [release notes](https://ggplot2.tidyverse.org/news/index.html#ggplot2-3-0-0).
+
+### `aes()` contains quosures
+
+The primary developer-facing change is that `aes()` now contains [quosures](https://adv-r.hadley.nz/evaluation.html#quosures) (expression + environment pairs), rather than symbols. As a result, you'll need to take a different approach to extracting the information you need. 
+
+```{r aes-quo}
+x_var <- quo(cyl)
+y_var <- quo(mpg)
+
+by_cyl <- mtcars %>%
+  group_by(!!x_var) %>%
+  summarise(mean = mean(!!y_var))
+
+ggplot(by_cyl, aes(!!x_var, mean)) +
+  geom_point()
+```
+
+You can now use [quasiquotation](https://adv-r.hadley.nz/quasiquotation.html) in [`aes()`](https://ggplot2.tidyverse.org/reference/aes.html), [`facet_wrap()`](https://ggplot2.tidyverse.org/reference/facet_wrap.html), and [`facet_grid()`](https://ggplot2.tidyverse.org/reference/facet_grid.html). For `aes()`, quasiquotation (`!!`, `!!!`, `:=`) replaces [`aes_()`](https://ggplot2.tidyverse.org/reference/aes_.html) and [`aes_string()`](https://ggplot2.tidyverse.org/reference/aes_.html) (though these functions are being soft deprecated, and will be around for a while).
+
+### Facetting with `vars()`
+
+To support quasiquotation in facetting we’ve added a new helper function: [`vars()`](https://ggplot2.tidyverse.org/reference/vars.html), short for variables. Instead of `facet_grid(x + y ~ a + b)` you can now write `facet_grid(vars(x, y), vars(a, b))`. The formula interface won’t go away; but the new `vars()` interface should be much easier to program with.
```

Comment:
"should be much easier to program with" ->  "supports tidy evaluation, so can be easily programmed with"